lib/scholar/linear/logistic_regression.ex

```diff
@@ -0,0 +1,193 @@
+defmodule Scholar.Linear.LogisticRegression do
+  @moduledoc """
+  Logistic regression in both binary and multinomial variants.
+  """
+  import Nx.Defn
+
+  @derive {Nx.Container, containers: [:coefficients, :bias], keep: [:mode]}
+  defstruct [:coefficients, :bias, :mode]
+
+  @doc """
+  Fits a logistic regression model for sample inputs `x` and sample
+  targets `y`.
+
+  Depending on number of classes the function chooses either binary
+  or multinomial logistic regression.
+  """
+  defn fit(x, y, opts \\ []) do
+    opts =
+      keyword!(opts,
+        iterations: 1000,
+        lr: 0.01,
+        num_classes: 1
```

Comment:
Maybe we should require the user to pass `:num_classes` as in metrics, otherwise the default only works for specific `y`s, wdyt?

Reply:
Sure, but how to require a field in the opts? ```:num_classes``` needs to be passed by the opts

Reply:
Same as in `Metrics`, we just raise if not present.